samples/simple_GAN.py

```diff
@@ -0,0 +1,302 @@
+
+        self.device = c.device
+
+        self.generator = c.generator
+        self.discriminator = c.discriminator
+        self.optimizer_G = c.optimizer_G
+        self.optimizer_D = c.optimizer_D
+
+        self.train_loader = c.train_loader
+        self.test_loader = c.test_loader
+
+        self.train_log_interval = c.train_log_interval
+        self.loop = c.training_loop
+        self.__is_log_parameters = c.is_log_parameters
+
+        self.image_directory = c.image_directory
+
+    def _train(self, epoch):
+        for i, (images, _) in logger.enum("Train", self.train_loader):
+            targets_real = (torch.FloatTensor(images.size(0), 1).uniform_(0.8, 1.0)).to(self.device)
+            targets_fake = (torch.FloatTensor(images.size(0), 1).uniform_(0.0, 0.2)).to(self.device)
+
+            images = images.to(self.device)
+
+            self.optimizer_D.zero_grad()
+            logits_real = self.discriminator(images)
+            fake_images = self.generator(noise(self.device, self.batch_size, self.noise_dim)).detach()
+            logits_fake = self.discriminator(fake_images)
+            discriminator_loss = DLoss(logits_real, logits_fake, targets_real, targets_fake)
+            discriminator_loss.backward()
+            self.optimizer_D.step()
+
+            self.optimizer_G.zero_grad()
+            fake_images = self.generator(noise(self.device, self.batch_size, self.noise_dim))
+            logits_fake = self.discriminator(fake_images)
+            generator_loss = GLoss(logits_fake, targets_real)
+            generator_loss.backward()
+            self.optimizer_G.step()
+
+            logger.store(G_Loss=generator_loss.item())
+            logger.store(D_Loss=discriminator_loss.item())
+            logger.add_global_step()
+
+            if i % self.train_log_interval == 0:
+                logger.write()
+
+        if epoch % 2 == 0:
+            viz_batch = fake_images.data.cpu().numpy()
+            fig = plt.figure(figsize=(8, 10))
+            for i in np.arange(1, 10):
+                ax = fig.add_subplot(3, 3, i)
+                img = viz_batch[i].squeeze()
+                plt.imshow(img)
+            plt.savefig(self.image_directory.format(epoch))
+            plt.show()
+
+    def __call__(self):
+        for i in self.loop:
+            self._train(i)
+
+
+class LoaderConfigs(configs.Configs):
+    train_loader: torch.utils.data.DataLoader
+    test_loader: torch.utils.data.DataLoader
+
+
+class Configs(training_loop.TrainingLoopConfigs, LoaderConfigs):
+    epochs: int = 50
+
+    loop_step = 'loop_step'
+    loop_count = 'loop_count'
+
+    is_save_models = True
+    batch_size: int = 128
+    test_batch_size: int = 128
+
+    use_cuda: float = True
+    cuda_device: int = 0
+    seed: int = 5
+    train_log_interval: int = 10
+
+    is_log_parameters: bool = True
+
+    device: any
+
+    generator = 'set_generator'
+    discriminator = 'set_discriminator'
+
+    optimizer_G = 'generator_optimizer'
```

Comment:
No need to specify the options if there's only one option available.
`optimizer_G:  Optimizer` should be enough